slow_wsgi_logger/tests/test_slow_wsgi_logger.py

```diff
@@ -0,0 +1,190 @@
+        # assertNoLogs would be better here but present from python 3.10 and greater
+        with self.assertLogs("odoo.addons.slow_wsgi_logger", level="DEBUG") as cm:
+            self.url_open(f"/waiting?wait={wait}", allow_redirects=False)
+            # log something in debug juste to avoid raising because nothing logged
+            _logger.debug("the only expected log")
+        self.assertEqual(len(cm.output), 1, cm.output)
+        self.assertEqual(cm.records[0].levelno, logging.DEBUG)
+        self.assertEqual(cm.records[0].msg, "the only expected log")
+
+
+@tagged("post_install", "-at_install")
+class TestSlowWSGILoggerJSONRPCRequest(HttpSavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.min_duration = 0.1
+        cls.level = "WARN"
+        config = {}
+        config["log_wsgi_request_min_duration"] = str(cls.min_duration)
+        config["log_wsgi_request_level"] = cls.level
+
+        from odoo.http import HttpRequest, JsonRequest
+
+        previous_http_request_dispatch = HttpRequest.dispatch
+        previous_json_request_dispatch = JsonRequest.dispatch
+
+        initialize_slow_wgsi_request_logger(config)
+
+        def restore_dispatch():
+            HttpRequest.dispatch = previous_http_request_dispatch
+            JsonRequest.dispatch = previous_json_request_dispatch
+
+        cls.addClassCleanup(restore_dispatch)
+
+        cls.loader = FakeModelLoader(cls.env, cls.__module__)
+        cls.addClassCleanup(cls.loader.restore_registry)
+
+        cls.loader.backup_registry()
+        from .fake_test_model import FakeTestModel
+
+        cls.loader.update_registry((FakeTestModel,))
+
+    def setUp(self):
+        super().setUp()
+        self.admin_uid = self.env.ref("base.user_admin").id
+        self.db_name = common.get_db_name()
+        self.uid = self.xmlrpc_common.login(self.db_name, "admin", "admin")
+        self.json_rpc_url = "http://%s:%s/jsonrpc" % (
+            common.HOST,
+            odoo.tools.config["http_port"],
+        )
+
+    def tearDown(self):
+        super().tearDown()
+        self.loader.restore_registry()
+
+    def _json_call(self, *args, headers=None):
+        return self.opener.post(
+            self.json_rpc_url,
+            json={
+                "jsonrpc": "2.0",
+                "id": None,
+                "method": "call",
+                "params": {"service": "object", "method": "execute", "args": args},
+            },
+            allow_redirects=False,
+            headers=headers,
+        )
+
+    def test_slow_name_search_should_logged(self):
+        wait = self.min_duration + 0.1
+        with self.assertLogs("odoo.addons.slow_wsgi_logger", level=self.level) as cm:
+            self._json_call(
+                self.db_name,
+                self.admin_uid,
+                "admin",
+                "fake.test",
+                "name_search",
+                wait,
+            )
```

Comment:
Test is [failing](https://github.com/OCA/server-tools/actions/runs/8802245846/job/24157487806#step:8:1891). Not quite sure what to make of it, to my best interpretation skills the `fake.test` model is not properly loaded, hence test fails and Odoo attempts to restore the savepoint but can't find it and the cursor is reset to an unusable state. The next query that has to do with registry restore, then also fails to execute.